### ISSUE TITLE
Downloads not working

### ISSUE THREAD

**Issue by partchimp** [Ascoware/get-iplayer-automator#399]
Trying to use GiA today for the first time in a couple of months and it won't download anything I've tried (all BBC). When I hit 'start', the status changes to 'waiting', and nothing happens. No error message, just nothing.

I'm in the UK so never had to use a VPN, but I tried this just in case and the result was the same.

I'm using v1.23.0 on MacOS 12.3.1

I'm trying to download this programme:
https://www.bbc.co.uk/iplayer/episode/p00gj4k1/the-century-of-the-self-4-eight-people-sipping-wine-in-kettering

I've also tried deleting and reinstalling, but still no joy.

Log file is attached.
[log.txt](https://github.com/Ascoware/get-iplayer-automator/files/8482327/log.txt)

**Comment by mike19132** [comment]
Similar problem, but when starting the download it hangs 'starting download' both BBc and ITV

**Comment by skovatch** [comment]
To diagnose it further I need a log with verbose mode turned on. Open up Preferences and check "Verbose Mode", then save the output of the log.

**Comment by partchimp** [comment]
Thanks - here's an updated log file.
Not that now the status doesn't even change to 'waiting' when I try to start the download - it just stays at 'available'.
[GIA verbose log.txt](https://github.com/Ascoware/get-iplayer-automator/files/8549621/GIA.verbose.log.txt)

**Comment by mike19132** [comment]
ITV programme: 'available' then 'starting download' then fails.
Verbose log attached

Get iPlayer Automator 1.23.0 Initialized.
<AppController: 0x7fccb0305ca0>: Updating Programme Index Feeds...

/Applications/Get iPlayer
Automator.app/Contents/Resources/get_iplayer/perl/bin/get_iplayer
-e3600
--type=tv
--refresh
--nopurge
--profile-dir=/Users/Mike/Library/Application Support/Get iPlayer Automator
.*
INFO: Indexing tv programmes (concurrent)
INFO: Added 193 tv programmes to cache
INFO: 7594 matching programmes
<AppController: 0x7fccb0305ca0>: Index Updated.


INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
AppController: Starting Downloads


Downloading Show 1/1:

Downloading The Thief, His Wife and The Canoe: Series 1
INFO: Preparing Request for Auth Info
INFO: Requesting Metadata.
DEBUG: Metadata response status code: 200
INFO: Metadata processed.
DEBUG: youtube-dl args:["
https://www.itv.com/hub/the-thief-his-wife-and-the-canoe/10a1187a0001",
"--user-agent", "Mozilla/5.0", "-f", "mp4/best", "-o",
"/Users/Mike/Desktop/The Thief, His Wife and The Canoe/The Thief, His Wife
and The Canoe.s01e01.Episode 1.%(ext)s", "--verbose"]
DEBUG: youtube-dl environment: ["SSL_CERT_FILE": "/Applications/Get iPlayer
Automator.app/Contents/Resources/cacert.pem", "PATH": "/Applications/Get
iPlayer Automator.app/Contents/Resources/yt-dlp_macos:/Applications/Get
iPlayer Automator.app/Contents/Resources/get_iplayer/utils/bin"]
[debug] Command-line config: ['
https://www.itv.com/hub/the-thief-his-wife-and-the-canoe/10a1187a0001',
'--user-agent', 'Mozilla/5.0', '-f', 'mp4/best', '-o',
'/Users/Mike/Desktop/The Thief, His Wife and The Canoe/The Thief, His Wife
and The Canoe.s01e01.Episode 1.%(ext)s', '--verbose']
[debug] Encodings: locale US-ASCII, fs utf-8, out ascii (No ANSI), err
ascii (No ANSI), pref US-ASCII
[debug] yt-dlp version 2021.12.27 [6223f67a8] (mac_dir)
[debug] Python version 3.9.9 (CPython 64bit) - macOS-12.3.1-x86_64-64bit
[debug] exe versions: ffmpeg 4.3.2-tessus
[debug] Optional libraries: Cryptodome, sqlite
[debug] Proxy map: {}
[debug] Using fake IP 25.141.198.198 (GB) as X-Forwarded-For
[debug] [ITV] Extracting URL:
https://www.itv.com/hub/the-thief-his-wife-and-the-canoe/10a1187a0001
[ITV] 10a1187a0001: Downloading webpage
ERROR: [ITV] 10a1187a0001: Unable to download webpage: The read operation
timed out (caused by timeout('The read operation timed out')); please
report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are
using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to
update. Be sure to call yt-dlp with the --verbose flag and include its
complete output. (caused by timeout('The read operation timed out'));
please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure
you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on
how to update. Be sure to call yt-dlp with the --verbose flag and include
its complete output.
  File "yt_dlp/extractor/common.py", line 717, in _request_webpage
  File "yt_dlp/YoutubeDL.py", line 3458, in urlopen
  File "urllib/request.py", line 517, in open
  File "urllib/request.py", line 534, in _open
  File "urllib/request.py", line 494, in _call_chain
  File "yt_dlp/utils.py", line 1392, in https_open
  File "urllib/request.py", line 1350, in do_open
  File "http/client.py", line 1377, in getresponse
  File "http/client.py", line 320, in begin
  File "http/client.py", line 281, in _read_status
  File "socket.py", line 704, in readinto
  File "ssl.py", line 1241, in recv_into
  File "ssl.py", line 1099, in read
youtube-dl finished downloading
AppController: Downloads Finished

 Mike
 Sankamphaeng
 Chiang Mai
 Thailand

**Comment by arijitsadhu** [comment]
I am still getting the same problem downloading any BBC shows and gets stuck on "Waiting" after pressing start. The verbose log does not give anything meaningful:

Get iPlayer Automator 1.23.0 Initialized.
GetITVShows: ITV Cache Update Starting 
<AppController: 0x7fbd74f0f270>: Updating Programme Index Feeds...

/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer/perl/bin/get_iplayer
-e14400
--type=tv
--refresh
--nopurge
--profile-dir=/Users/arijitsadhu/Library/Application Support/Get iPlayer Automator
.*
GetITVShows (Info): Programmes: 553
INFO: Indexing tv programmes (concurrent)
INFO: Added 404 tv programmes to cache
INFO: 7553 matching programmes
GetITVShows (Info): Episodes: 7449
GetITVShows: Update Finished
<AppController: 0x7fbd74f0f270>: Index Updated.


INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
AppController: Starting Downloads

**Comment by mike19132** [comment]
I restored BBC downloads by using an earlier version of GIT that cleared
the problem, then  updated to 1.23.0 which then worked. But that did not
clear the ITV problem when it fails.It says 'finished downloading but has
not started I get a 'download failed' message. Verbose log attached
Get iPlayer Automator 1.23.0 Initialized.
<AppController: 0x7f838b726a00>: Updating Programme Index Feeds...

INFO: Indexing tv programmes (concurrent)
INFO: Added 190 tv programmes to cache
INFO: 7711 matching programmes
<AppController: 0x7f838b726a00>: Index Updated.


INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
AppController: Starting Downloads


Downloading Show 1/1:

Downloading The Thief, His Wife and The Canoe: Series 1
INFO: Preparing Request for Auth Info
INFO: Metadata processed.
youtube-dl finished downloading
AppController: Downloads Finished

 Mike
 Sankamphaeng
 Chiang Mai
 Thailand
